connectedDebugAndroidTest fails on API > 22

#### Software and hardware versions 
Collect v1.17.x

#### Problem description
`./gradlew connectedDebugAndroidTest` results in failed tests.

I've tried on an Nexus 5X API 26 emulator and a Moto G5 running Android 7

#### Expected behavior

#### Other information 
```
Starting 82 tests on Nexus_5X_API_26(AVD) - 8.0.0

org.odk.collect.android.GuidanceHintFormTest > guidanceVisibilityContentTest[Nexus_5X_API_26(AVD) - 8.0.0] SKIPPED

org.odk.collect.android.ResetAppStateTestCase > resetSettingsTest[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.AssertionError
        at org.junit.Assert.fail(Assert.java:86)

org.odk.collect.android.ResetAppStateTestCase > resetOSMDroidTest[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.AssertionError
        at org.junit.Assert.fail(Assert.java:86)

org.odk.collect.android.ResetAppStateTestCase > resetCacheTest[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.io.IOException: No such file or directory
        at java.io.UnixFileSystem.createFileExclusively0(Native Method)

org.odk.collect.android.ResetAppStateTestCase > resetLayersTest[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.io.IOException: No such file or directory
        at java.io.UnixFileSystem.createFileExclusively0(Native Method)

org.odk.collect.android.ResetAppStateTestCase > resetInstancesTest[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.AssertionError
        at org.junit.Assert.fail(Assert.java:86)

org.odk.collect.android.ResetAppStateTestCase > resetFormsTest[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.io.IOException: No such file or directory
        at java.io.UnixFileSystem.createFileExclusively0(Native Method)

org.odk.collect.android.SmsSubmissionMemoryTest > testMemoryConsumption[Nexus_5X_API_26(AVD) - 8.0.0] SKIPPED

org.odk.collect.android.dao.FormsDaoTest > updateInstanceTest[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.io.IOException: No such file or directory
        at java.io.UnixFileSystem.createFileExclusively0(Native Method)

org.odk.collect.android.dao.FormsDaoTest > getFormsCursorForFormIdTest[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.io.IOException: No such file or directory
        at java.io.UnixFileSystem.createFileExclusively0(Native Method)

org.odk.collect.android.dao.FormsDaoTest > getFormMediaPathTest[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.io.IOException: No such file or directory
        at java.io.UnixFileSystem.createFileExclusively0(Native Method)

org.odk.collect.android.dao.FormsDaoTest > getAllFormsCursorTest[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.io.IOException: No such file or directory
        at java.io.UnixFileSystem.createFileExclusively0(Native Method)

org.odk.collect.android.dao.FormsDaoTest > getFormsCursorForFormFilePathTest[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.io.IOException: No such file or directory
        at java.io.UnixFileSystem.createFileExclusively0(Native Method)

org.odk.collect.android.dao.FormsDaoTest > getFormsCursorTest[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.io.IOException: No such file or directory
        at java.io.UnixFileSystem.createFileExclusively0(Native Method)

org.odk.collect.android.dao.InstancesDaoTest > getUnsentInstancesCursorTest[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.AssertionError: expected:<4> but was:<0>
        at org.junit.Assert.fail(Assert.java:88)

org.odk.collect.android.dao.InstancesDaoTest > updateInstanceTest[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.AssertionError: expected:<1> but was:<0>
        at org.junit.Assert.fail(Assert.java:88)

org.odk.collect.android.dao.InstancesDaoTest > getInstancesCursorForFilePathTest[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.AssertionError: expected:<1> but was:<0>
        at org.junit.Assert.fail(Assert.java:88)

org.odk.collect.android.dao.InstancesDaoTest > getSentInstancesCursorTest[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.AssertionError: expected:<2> but was:<0>
        at org.junit.Assert.fail(Assert.java:88)

org.odk.collect.android.dao.InstancesDaoTest > getFinalizedInstancesCursorTest[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.AssertionError: expected:<1> but was:<0>
        at org.junit.Assert.fail(Assert.java:88)

org.odk.collect.android.dao.InstancesDaoTest > getSavedInstancesCursorTest[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.AssertionError: expected:<5> but was:<0>
        at org.junit.Assert.fail(Assert.java:88)

org.odk.collect.android.dao.InstancesDaoTest > getInstancesCursorForIdTest[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.AssertionError: expected:<1> but was:<0>
        at org.junit.Assert.fail(Assert.java:88)

org.odk.collect.android.dao.InstancesDaoTest > getAllCompletedUndeletedInstancesCursorTest[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.AssertionError: expected:<2> but was:<0>
        at org.junit.Assert.fail(Assert.java:88)

org.odk.collect.android.tasks.DownloadFormListTaskTest > shouldProcessAndReturnAFormList[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.NullPointerException: Attempt to invoke interface method 'int android.database.Cursor.getCount()' on a null object reference
        at org.odk.collect.android.utilities.DownloadFormListUtils.isNewerFormVersionAvailable(DownloadFormListUtils.java:423)

org.odk.collect.android.tasks.InstanceServerUploaderTest > shouldUploadAnInstance[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.io.IOException: No such file or directory
        at java.io.UnixFileSystem.createFileExclusively0(Native Method)

org.odk.collect.android.utilities.BitmapScaledToDisplayTest > scaleDownBitmapWhenNeeded[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.AssertionError: expected:<500> but was:<333>
        at org.junit.Assert.fail(Assert.java:88)

org.odk.collect.android.utilities.CustomSQLiteQueryBuilderTestCase > addColumnTest[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        android.database.sqlite.SQLiteCantOpenDatabaseException: unknown error (code 14): Could not open database
        at android.database.sqlite.SQLiteConnection.nativeOpen(Native Method)

org.odk.collect.android.utilities.CustomSQLiteQueryBuilderTestCase > dropTableTest[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        android.database.sqlite.SQLiteCantOpenDatabaseException: unknown error (code 14): Could not open database
        at android.database.sqlite.SQLiteConnection.nativeOpen(Native Method)

org.odk.collect.android.utilities.CustomSQLiteQueryBuilderTestCase > copyTableTest[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        android.database.sqlite.SQLiteCantOpenDatabaseException: unknown error (code 14): Could not open database
        at android.database.sqlite.SQLiteConnection.nativeOpen(Native Method)

org.odk.collect.android.utilities.CustomSQLiteQueryBuilderTestCase > renameTableTest[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        android.database.sqlite.SQLiteCantOpenDatabaseException: unknown error (code 14): Could not open database
        at android.database.sqlite.SQLiteConnection.nativeOpen(Native Method)

org.odk.collect.android.utilities.FormIndexSavepointTest > saveAndReadFormIndexTest[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.AssertionError: expected:<-1, > but was:<null>
        at org.junit.Assert.fail(Assert.java:88)

org.odk.collect.android.utilities.ImageConverterTest > scaleImageDownFormAndSettingsLevel1[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.NullPointerException: Attempt to invoke virtual method 'int android.graphics.Bitmap.getWidth()' on a null object reference
        at org.odk.collect.android.utilities.ImageConverterTest.scaleImageDownFormAndSettingsLevel1(ImageConverterTest.java:261)

org.odk.collect.android.utilities.ImageConverterTest > scaleImageDownFormAndSettingsLevel2[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.NullPointerException: Attempt to invoke virtual method 'int android.graphics.Bitmap.getWidth()' on a null object reference
        at org.odk.collect.android.utilities.ImageConverterTest.scaleImageDownFormAndSettingsLevel2(ImageConverterTest.java:272)

org.odk.collect.android.utilities.ImageConverterTest > scaleImageDownFormLevelOnly1[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.NullPointerException: Attempt to invoke virtual method 'int android.graphics.Bitmap.getWidth()' on a null object reference
        at org.odk.collect.android.utilities.ImageConverterTest.scaleImageDownFormLevelOnly1(ImageConverterTest.java:74)

org.odk.collect.android.utilities.ImageConverterTest > scaleImageDownFormLevelOnly2[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.NullPointerException: Attempt to invoke virtual method 'int android.graphics.Bitmap.getWidth()' on a null object reference
        at org.odk.collect.android.utilities.ImageConverterTest.scaleImageDownFormLevelOnly2(ImageConverterTest.java:85)

org.odk.collect.android.utilities.ImageConverterTest > scaleImageDownFormLevelOnly3[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.NullPointerException: Attempt to invoke virtual method 'int android.graphics.Bitmap.getWidth()' on a null object reference
        at org.odk.collect.android.utilities.ImageConverterTest.scaleImageDownFormLevelOnly3(ImageConverterTest.java:96)

org.odk.collect.android.utilities.ImageConverterTest > scaleImageDownFormLevelOnly4[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.NullPointerException: Attempt to invoke virtual method 'int android.graphics.Bitmap.getWidth()' on a null object reference
        at org.odk.collect.android.utilities.ImageConverterTest.scaleImageDownFormLevelOnly4(ImageConverterTest.java:107)

org.odk.collect.android.utilities.ImageConverterTest > scaleImageDownFormLevelOnly5[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.NullPointerException: Attempt to invoke virtual method 'int android.graphics.Bitmap.getWidth()' on a null object reference
        at org.odk.collect.android.utilities.ImageConverterTest.scaleImageDownFormLevelOnly5(ImageConverterTest.java:118)

org.odk.collect.android.utilities.ImageConverterTest > scaleImageDownFormLevelOnly6[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.NullPointerException: Attempt to invoke virtual method 'int android.graphics.Bitmap.getWidth()' on a null object reference
        at org.odk.collect.android.utilities.ImageConverterTest.scaleImageDownFormLevelOnly6(ImageConverterTest.java:129)

org.odk.collect.android.utilities.ImageConverterTest > scaleImageDownFormLevelOnly7[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.NullPointerException: Attempt to invoke virtual method 'int android.graphics.Bitmap.getWidth()' on a null object reference
        at org.odk.collect.android.utilities.ImageConverterTest.scaleImageDownFormLevelOnly7(ImageConverterTest.java:140)

org.odk.collect.android.utilities.ImageConverterTest > scaleImageDownFormLevelOnly8[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.NullPointerException: Attempt to invoke virtual method 'int android.graphics.Bitmap.getWidth()' on a null object reference
        at org.odk.collect.android.utilities.ImageConverterTest.scaleImageDownFormLevelOnly8(ImageConverterTest.java:151)

org.odk.collect.android.utilities.ImageConverterTest > scaleImageDownFormLevelOnly9[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.NullPointerException: Attempt to invoke virtual method 'int android.graphics.Bitmap.getWidth()' on a null object reference
        at org.odk.collect.android.utilities.ImageConverterTest.scaleImageDownFormLevelOnly9(ImageConverterTest.java:162)

org.odk.collect.android.utilities.ImageConverterTest > scaleImageDownSettingsLevelOnly1[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.NullPointerException: Attempt to invoke virtual method 'int android.graphics.Bitmap.getWidth()' on a null object reference
        at org.odk.collect.android.utilities.ImageConverterTest.scaleImageDownSettingsLevelOnly1(ImageConverterTest.java:206)

org.odk.collect.android.utilities.ImageConverterTest > scaleImageDownSettingsLevelOnly2[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.NullPointerException: Attempt to invoke virtual method 'int android.graphics.Bitmap.getWidth()' on a null object reference
        at org.odk.collect.android.utilities.ImageConverterTest.scaleImageDownSettingsLevelOnly2(ImageConverterTest.java:217)

org.odk.collect.android.utilities.ImageConverterTest > scaleImageDownSettingsLevelOnly3[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.NullPointerException: Attempt to invoke virtual method 'int android.graphics.Bitmap.getWidth()' on a null object reference
        at org.odk.collect.android.utilities.ImageConverterTest.scaleImageDownSettingsLevelOnly3(ImageConverterTest.java:228)

org.odk.collect.android.utilities.ImageConverterTest > scaleImageDownSettingsLevelOnly4[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.NullPointerException: Attempt to invoke virtual method 'int android.graphics.Bitmap.getWidth()' on a null object reference
        at org.odk.collect.android.utilities.ImageConverterTest.scaleImageDownSettingsLevelOnly4(ImageConverterTest.java:239)

org.odk.collect.android.utilities.ImageConverterTest > scaleImageDownSettingsLevelOnly5[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.NullPointerException: Attempt to invoke virtual method 'int android.graphics.Bitmap.getWidth()' on a null object reference
        at org.odk.collect.android.utilities.ImageConverterTest.scaleImageDownSettingsLevelOnly5(ImageConverterTest.java:250)

org.odk.collect.android.utilities.ImageConverterTest > scaleImageDownFormLevelOnly10[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.NullPointerException: Attempt to invoke virtual method 'int android.graphics.Bitmap.getWidth()' on a null object reference
        at org.odk.collect.android.utilities.ImageConverterTest.scaleImageDownFormLevelOnly10(ImageConverterTest.java:173)

org.odk.collect.android.utilities.ImageConverterTest > scaleImageDownFormLevelOnly11[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.NullPointerException: Attempt to invoke virtual method 'int android.graphics.Bitmap.getWidth()' on a null object reference
        at org.odk.collect.android.utilities.ImageConverterTest.scaleImageDownFormLevelOnly11(ImageConverterTest.java:184)

org.odk.collect.android.utilities.ImageConverterTest > scaleImageDownFormLevelOnly12[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.NullPointerException: Attempt to invoke virtual method 'int android.graphics.Bitmap.getWidth()' on a null object reference
        at org.odk.collect.android.utilities.ImageConverterTest.scaleImageDownFormLevelOnly12(ImageConverterTest.java:195)

org.odk.collect.android.utilities.ImageConverterTest > rotateImage1[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.NullPointerException: Attempt to invoke virtual method 'int android.graphics.Bitmap.getWidth()' on a null object reference
        at org.odk.collect.android.utilities.ImageConverterTest.rotateImage1(ImageConverterTest.java:283)

org.odk.collect.android.utilities.ImageConverterTest > rotateImage2[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.NullPointerException: Attempt to invoke virtual method 'int android.graphics.Bitmap.getWidth()' on a null object reference
        at org.odk.collect.android.utilities.ImageConverterTest.rotateImage2(ImageConverterTest.java:294)

org.odk.collect.android.utilities.ImageConverterTest > rotateImage3[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.NullPointerException: Attempt to invoke virtual method 'int android.graphics.Bitmap.getWidth()' on a null object reference
        at org.odk.collect.android.utilities.ImageConverterTest.rotateImage3(ImageConverterTest.java:305)

org.odk.collect.android.utilities.ImageConverterTest > rotateImage4[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.NullPointerException: Attempt to invoke virtual method 'int android.graphics.Bitmap.getWidth()' on a null object reference
        at org.odk.collect.android.utilities.ImageConverterTest.rotateImage4(ImageConverterTest.java:316)

org.odk.collect.android.utilities.ImageConverterTest > rotateImage5[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.NullPointerException: Attempt to invoke virtual method 'int android.graphics.Bitmap.getWidth()' on a null object reference
        at org.odk.collect.android.utilities.ImageConverterTest.rotateImage5(ImageConverterTest.java:327)

org.odk.collect.android.utilities.ImageConverterTest > executeConversionWithoutAnySettings[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.NullPointerException: Attempt to invoke virtual method 'int android.graphics.Bitmap.getWidth()' on a null object reference
        at org.odk.collect.android.utilities.ImageConverterTest.executeConversionWithoutAnySettings(ImageConverterTest.java:63)

org.odk.collect.android.utilities.ImageConverterTest > rotateAndScaleDownImage[Nexus_5X_API_26(AVD) - 8.0.0] FAILED
        java.lang.NullPointerException: Attempt to invoke virtual method 'int android.graphics.Bitmap.getWidth()' on a null object reference
        at org.odk.collect.android.utilities.ImageConverterTest.rotateAndScaleDownImage(ImageConverterTest.java:338)
Tests on Nexus_5X_API_26(AVD) - 8.0.0 failed: Test run failed to complete. Expected 82 tests, received 81
```

So I noticed that when I run the app from Android Studio it initially asks for permissions after which the app launches, if I run `./gradlew connectedDebugAndroidTest` at this point only two tests fail `InstanceServerUploaderTest > shouldUploadAnInstance` & `BitmapScaledToDisplayTest > scaleDownBitmapWhenNeeded`  but if I run the tests without running the app first I get the same output as @yanokwa . So it does seem like a permissions issue but I am not very sure why the other two tests fail. I think it expected a device id but that's missing and for the second test `BitmapScaledToDisplayTest > scaleDownBitmapWhenNeeded` I think there might be some issue with `FileUtils.getBitmapScaledToDisplay` because the expected height doesn't match the height returned by this function, this test failed for @yanokwa also with the same values, although I am not sure what that implies.

```
Starting 82 tests on Moto E (4) Plus - 7.1.1

org.odk.collect.android.GuidanceHintFormTest > guidanceVisibilityContentTest[Moto E (4) Plus - 7.1.1] SKIPPED 

org.odk.collect.android.SmsSubmissionMemoryTest > testMemoryConsumption[Moto E (4) Plus - 7.1.1] SKIPPED 

org.odk.collect.android.tasks.InstanceServerUploaderTest > shouldUploadAnInstance[Moto E (4) Plus - 7.1.1] FAILED 
        java.lang.AssertionError: Expected </submission?deviceID=> to match </submission\?deviceID=\w+%3A\w+>.
        at org.odk.collect.android.test.TestUtils.assertMatches(TestUtils.java:97)

org.odk.collect.android.utilities.BitmapScaledToDisplayTest > scaleDownBitmapWhenNeeded[Moto E (4) Plus - 7.1.1] FAILED 
        java.lang.AssertionError: expected:<500> but was:<333>
        at org.junit.Assert.fail(Assert.java:88)

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':collect_app:connectedDebugAndroidTest'.
> There were failing tests. See the report at: file:///home/lola/collect/collect_app/build/reports/androidTests/connected/index.html

* Try:
Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':collect_app:connectedDebugAndroidTest'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:103)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:73)
        at org.gradle.api.internal.tasks.execution.OutputDirectoryCreatingTaskExecuter.execute(OutputDirectoryCreatingTaskExecuter.java:51)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:59)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:101)
        at org.gradle.api.internal.tasks.execution.FinalizeInputFilePropertiesTaskExecuter.execute(FinalizeInputFilePropertiesTaskExecuter.java:44)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:88)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter.java:248)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:241)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:230)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.processTask(DefaultTaskPlanExecutor.java:123)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.access$200(DefaultTaskPlanExecutor.java:79)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:104)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:98)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:623)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:578)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:98)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor.process(DefaultTaskPlanExecutor.java:59)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter.execute(DefaultTaskGraphExecuter.java:128)
        at org.gradle.execution.SelectedTaskExecutionAction.execute(SelectedTaskExecutionAction.java:37)
        at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
        at org.gradle.execution.DefaultBuildExecuter.access$000(DefaultBuildExecuter.java:23)
        at org.gradle.execution.DefaultBuildExecuter$1.proceed(DefaultBuildExecuter.java:43)
        at org.gradle.execution.DryRunBuildExecutionAction.execute(DryRunBuildExecutionAction.java:46)
        at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
        at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:30)
        at org.gradle.initialization.DefaultGradleLauncher$ExecuteTasks.run(DefaultGradleLauncher.java:318)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
        at org.gradle.initialization.DefaultGradleLauncher.runTasks(DefaultGradleLauncher.java:204)
        at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:134)
        at org.gradle.initialization.DefaultGradleLauncher.executeTasks(DefaultGradleLauncher.java:109)
        at org.gradle.internal.invocation.GradleBuildController$1.call(GradleBuildController.java:78)
        at org.gradle.internal.invocation.GradleBuildController$1.call(GradleBuildController.java:75)
        at org.gradle.internal.work.DefaultWorkerLeaseService.withLocks(DefaultWorkerLeaseService.java:152)
        at org.gradle.internal.invocation.GradleBuildController.doBuild(GradleBuildController.java:100)
        at org.gradle.internal.invocation.GradleBuildController.run(GradleBuildController.java:75)
        at org.gradle.tooling.internal.provider.ExecuteBuildActionRunner.run(ExecuteBuildActionRunner.java:28)
        at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
        at org.gradle.tooling.internal.provider.ValidatingBuildActionRunner.run(ValidatingBuildActionRunner.java:32)
        at org.gradle.launcher.exec.RunAsBuildOperationBuildActionRunner$1.run(RunAsBuildOperationBuildActionRunner.java:43)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
        at org.gradle.launcher.exec.RunAsBuildOperationBuildActionRunner.run(RunAsBuildOperationBuildActionRunner.java:40)
        at org.gradle.tooling.internal.provider.SubscribableBuildActionRunner.run(SubscribableBuildActionRunner.java:51)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:49)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:32)
        at org.gradle.launcher.exec.BuildTreeScopeBuildActionExecuter.execute(BuildTreeScopeBuildActionExecuter.java:39)
        at org.gradle.launcher.exec.BuildTreeScopeBuildActionExecuter.execute(BuildTreeScopeBuildActionExecuter.java:25)
        at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:80)
        at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:53)
        at org.gradle.tooling.internal.provider.ServicesSetupBuildActionExecuter.execute(ServicesSetupBuildActionExecuter.java:57)
        at org.gradle.tooling.internal.provider.ServicesSetupBuildActionExecuter.execute(ServicesSetupBuildActionExecuter.java:32)
        at org.gradle.tooling.internal.provider.GradleThreadBuildActionExecuter.execute(GradleThreadBuildActionExecuter.java:36)
        at org.gradle.tooling.internal.provider.GradleThreadBuildActionExecuter.execute(GradleThreadBuildActionExecuter.java:25)
        at org.gradle.tooling.internal.provider.ParallelismConfigurationBuildActionExecuter.execute(ParallelismConfigurationBuildActionExecuter.java:43)
        at org.gradle.tooling.internal.provider.ParallelismConfigurationBuildActionExecuter.execute(ParallelismConfigurationBuildActionExecuter.java:29)
        at org.gradle.tooling.internal.provider.StartParamsValidatingActionExecuter.execute(StartParamsValidatingActionExecuter.java:64)
        at org.gradle.tooling.internal.provider.StartParamsValidatingActionExecuter.execute(StartParamsValidatingActionExecuter.java:29)
        at org.gradle.tooling.internal.provider.SessionFailureReportingActionExecuter.execute(SessionFailureReportingActionExecuter.java:59)
        at org.gradle.tooling.internal.provider.SessionFailureReportingActionExecuter.execute(SessionFailureReportingActionExecuter.java:44)
        at org.gradle.tooling.internal.provider.SetupLoggingActionExecuter.execute(SetupLoggingActionExecuter.java:45)
        at org.gradle.tooling.internal.provider.SetupLoggingActionExecuter.execute(SetupLoggingActionExecuter.java:30)
        at org.gradle.launcher.daemon.server.exec.ExecuteBuild.doBuild(ExecuteBuild.java:67)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:122)
        at org.gradle.launcher.daemon.server.exec.WatchForDisconnection.execute(WatchForDisconnection.java:37)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:122)
        at org.gradle.launcher.daemon.server.exec.ResetDeprecationLogger.execute(ResetDeprecationLogger.java:26)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:122)
        at org.gradle.launcher.daemon.server.exec.RequestStopIfSingleUsedDaemon.execute(RequestStopIfSingleUsedDaemon.java:34)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:122)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:74)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:72)
        at org.gradle.util.Swapper.swap(Swapper.java:38)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput.execute(ForwardClientInput.java:72)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:122)
        at org.gradle.launcher.daemon.server.exec.LogAndCheckHealth.execute(LogAndCheckHealth.java:55)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:122)
        at org.gradle.launcher.daemon.server.exec.LogToClient.doBuild(LogToClient.java:62)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:122)
        at org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment.doBuild(EstablishBuildEnvironment.java:82)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:122)
        at org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy$1.run(StartBuildOrRespondWithBusy.java:50)
        at org.gradle.launcher.daemon.server.DaemonStateCoordinator$1.run(DaemonStateCoordinator.java:295)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: org.gradle.api.GradleException: There were failing tests. See the report at: file:///home/lola/collect/collect_app/build/reports/androidTests/connected/index.html
        at com.android.build.gradle.internal.tasks.DeviceProviderInstrumentTestTask.runTests(DeviceProviderInstrumentTestTask.java:184)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:46)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:39)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:26)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:788)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:755)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:124)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:199)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:110)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:113)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:95)
        ... 104 more


* Get more help at https://help.gradle.org

BUILD FAILED in 3m 56s
60 actionable tasks: 10 executed, 50 up-to-date
```

--- a/collect_app/src/androidTest/java/org/odk/collect/android/tasks/InstanceServerUploaderTaskTest.java
+++ b/collect_app/src/androidTest/java/org/odk/collect/android/tasks/InstanceServerUploaderTaskTest.java
@@ -1,9 +1,11 @@
 package org.odk.collect.android.tasks;
 
 import android.net.Uri;
+import android.support.test.rule.GrantPermissionRule;
 
 import org.junit.After;
 import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
 import org.odk.collect.android.dao.InstancesDao;
 import org.odk.collect.android.dto.Instance;
@@ -22,6 +24,10 @@ import static org.odk.collect.android.test.TestUtils.createTempFile;
 import static org.odk.collect.android.test.TestUtils.resetInstancesContentProvider;
 
 public class InstanceServerUploaderTaskTest extends MockedServerTest {
+
+    @Rule
+    public GrantPermissionRule runtimepermissionrule = GrantPermissionRule.grant(android.Manifest.permission.READ_PHONE_STATE);
+
     private InstancesDao dao;
 
     @Before

--- a/collect_app/src/androidTest/java/org/odk/collect/android/utilities/BitmapScaledToDisplayTest.java
+++ b/collect_app/src/androidTest/java/org/odk/collect/android/utilities/BitmapScaledToDisplayTest.java
@@ -24,7 +24,6 @@ public class BitmapScaledToDisplayTest {
         runScaleTest(1000,   1000,    500,    500,    500,    500,    false);
         runScaleTest( 600,    800,    600,    200,    150,    200,    false);
         runScaleTest( 500,    400,    250,    200,    250,    200,    false);
-        runScaleTest(2000,    800,    300,    400,    500,    200,    false);
     }
 
     @Test

